### PR TITLE
nix: fix aarch64 build

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
At IRC, .:Ke:., asked "shouldn't that be stdenv.isx86_64 && is24"[?]

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -55,7 +55,8 @@ common =
         ]
         ++ lib.optionals stdenv.isDarwin [ Security ]
         ++ lib.optional (stdenv.isLinux || stdenv.isDarwin) libsodium
-        ++ lib.optionals is24 [ libarchive gmock lowdown libcpuid ]
+        ++ lib.optionals is24 [ libarchive gmock lowdown ]
+        ++ lib.optional (stdenv.isx86_64) libcpuid
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix/default.nix:59]:
```suggestion
        ++ lib.optional (is24 && stdenv.isx86_64) libcpuid
```

**Review by aforemny — APPROVED** [review]

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (aforemny) [reply pkgs/tools/package-management/nix/default.nix]: Woops. It seems I left this open before this was fixed by @SuperSandro2000 . Sorry for the noise.

**Review by shlevy — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116032` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>nix-exec</li>
</ul>
</details>
<details>
<summary>
2 packages blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
<li>tests.nixos-functions.nixosTest-test</li>
</ul>
</details>
<details>

  <summary>3 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>haskellPackages.hercules-ci-agent: log was empty</li>
<li>hercules-ci-agent: log was empty</li>
<li>libnixxml: log was empty</li>
  </ul>
</details>
<details>
<summary>
46 packages built:
</summary>
<ul>
<li>bundix</li>
<li>cabal2nix</li>
<li>cachix</li>
<li>common-updater-scripts</li>
<li>crate2nix</li>
<li>crystal2nix</li>
<li>dep2nix</li>
<li>fusionInventory</li>
<li>go2nix</li>
<li>haskellPackages.cachix</li>
<li>haskellPackages.hercules-ci-cnix-store</li>
<li>haskellPackages.nix-paths</li>
<li>haskellPackages.update-nix-fetchgit</li>
<li>lispPackages.quicklisp-to-nix</li>
<li>lispPackages.quicklisp-to-nix-system-info</li>
<li>nix (nixStable)</li>
<li>nix-bundle</li>
<li>nix-doc</li>
<li>nix-du</li>
<li>nix-index</li>
<li>nix-pin</li>
<li>nix-plugins</li>
<li>nix-prefetch</li>
<li>nix-prefetch-bzr</li>
<li>nix-prefetch-cvs</li>
<li>nix-prefetch-docker</li>
<li>nix-prefetch-git</li>
<li>nix-prefetch-hg</li>
<li>nix-prefetch-scripts</li>
<li>nix-prefetch-svn</li>
<li>nix-serve</li>
<li>nix-update</li>
<li>nix-update-source</li>
<li>nixos-generators</li>
<li>nixos-rebuild</li>
<li>nixos-shell</li>
<li>nixui</li>
<li>python38Packages.nix-kernel</li>
<li>python38Packages.nixpkgs</li>
<li>python38Packages.pythonix</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.nixpkgs</li>
<li>python39Packages.pythonix</li>
<li>update-nix-fetchgit</li>
<li>vgo2nix</li>
<li>vulnix</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Is that an unrelated build failure?

nix-build -A pkgsCross.aarch64-multiplatform.nix -A pkgsCross.aarch64-multiplatform.nixUnstable

```
  LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/lib/libnixexpr.so
  LD     src/libcmd/libnixcmd.so
  LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/lib/libnixcmd.so
  LD     src/nix/nix
  LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/bin/nix
  GEN    doc/manual/conf-file.json
/nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/bin/nix: line 1: syntax error: unexpected "|"
make: *** [doc/manual/local.mk:61: doc/manual/conf-file.json] Error 2
```

**Review by aforemny — APPROVED** [review]
LGTM

I successfully built nix for aarch64 using qemu emulation on x86_64.

**Comment by superherointj** [comment]
I have succesfuly built master for my RockPro64 (aarch64) (1hr ago) with proposed patch.

**Comment by Mic92** [comment]
> Is that an unrelated build failure?
> 
> nix-build -A pkgsCross.aarch64-multiplatform.nix -A pkgsCross.aarch64-multiplatform.nixUnstable
> 
> ```
>   LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/lib/libnixexpr.so
>   LD     src/libcmd/libnixcmd.so
>   LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/lib/libnixcmd.so
>   LD     src/nix/nix
>   LD     /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/bin/nix
>   GEN    doc/manual/conf-file.json
> /nix/store/5wajipic9r5nyis3imia7gz8xm71cczp-nix-2.4pre20210308_1c0e3e4-aarch64-unknown-linux-gnu/bin/nix: line 1: syntax error: unexpected "|"
> make: *** [doc/manual/local.mk:61: doc/manual/conf-file.json] Error 2
> ```

I think so

**Comment by cole-h** [comment]
Sounds similar / related to https://github.com/NixOS/nix/issues/4622.

**Comment by Pacman99** [comment]
Can this be back ported too, since the update PR broke nix on aarch64 20.09.

**Comment by Mic92** [comment]
Backport: https://github.com/NixOS/nixpkgs/pull/116083

**Pull request merged**